controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -778,20 +824,48 @@ func (r *StorageClassClaimReconciler) reconcileCephFilesystemSubVolumeGroup() er
 	}
 
 	var cephFileSystemName string
+	var dataPoolName string
 	availableCephFileSystems := len(cephFilesystemList.Items)
 	if availableCephFileSystems == 0 {
 		return fmt.Errorf("no CephFileSystem found in the cluster")
 	}
-	if availableCephFileSystems > 1 {
-		klog.Warningf("More than one CephFileSystem found in the cluster, selecting the first one")
-	}
 
-	cephFileSystemName = cephFilesystemList.Items[0].Name
+	cephClients := map[string]string{
+		"provisioner": r.cephClientProvisioner.Name,
+		"node":        r.cephClientNode.Name,
+	}
 
 	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephFilesystemSubVolumeGroup, func() error {
 		if err := r.own(r.cephFilesystemSubVolumeGroup); err != nil {
 			return err
 		}
+		deviceClass := r.deviceClass
+		if deviceClass == "" {
+			klog.Warning("No device class specified. Using the first CephFileSystem found in the cluster")
+			cephFileSystemName = cephFilesystemList.Items[0].Name
+			dataPoolName = cephFilesystemList.Items[0].Spec.DataPools[0].Name
+		} else {
+			for i := range cephFilesystemList.Items {
+				for j := range cephFilesystemList.Items[i].Spec.DataPools {
+					if cephFilesystemList.Items[i].Spec.DataPools[j].DeviceClass == deviceClass {
+						cephFileSystemName = cephFilesystemList.Items[i].Name
+						dataPoolName = cephFilesystemList.Items[i].Spec.DataPools[j].Name
+						break
+					}
+				}
+
+			}
+		}
```

Comment:
This is just complicated for no reason. This code only runs in a provided mode where we are guaranteed to have a single FS. If that assumption changes a lot of other things would break. In addition we know that device class is mandatory on profile so the other check is also non-relevant

Reply:
So the assumption is that we will have a single filesystem, even if we have multiple filesystems, we will always use the first one from the list operation? If yes i can remove this check

Reply:
As per the code we can have multiple filesystems(assumption in existing code) and we will assume the first in the list as the one to use but from ocs-operator will be deployed only one.